Feat/show drink hand

NickoLefrench opened the pull request:
HUD_Manager created and manages what drinks appears in the HUD during gameplay

BeerTap and CocktailMinigame have Sprites assigned to them that are fetched by the HUDManager. Everything is functional

The Alpha/Transparency of the image could be played with

Cleaned up some comments here and there as a side note

https://github.com/user-attachments/assets/6a3daa27-153f-451f-a824-a91ee527ad53

NickoLefrench committed:
Removed legacy code

removed chunk of code that was commented out

NickoLefrench committed:
Created HUD Manager

HUD Manager created to manage the Visual aspect of the players inventory

NickoLefrench committed:
Connecting HUD Drink to Inventory

HUD Manager and Player Inventory updated to change the HUD iumage during gameplay
CHanges in BeerTap and CocktailMinigame left commented out for now until sprites are done next

NickoLefrench committed:
Sprites for Inventory Sprites

NickoLefrench committed:
Finished HUD Inventory!

Player can see what drink they are holding in the top left corner
Adjusted comments in code in various spots for cleanliness
everything functional

vanyamil commented on Assets/Scripts/Inventory/PlayerInventory.cs:
```diff
@@ -16,6 +16,10 @@ public class PlayerInventory : MonoBehaviour
 
+        // Event to notify when a drink is added or removed
+        public delegate void DrinkChanged();
+        public event DrinkChanged OnDrinkChanged;
```
You don't have to make your own type, you can see examples of events in managers, they just use Action<>

NickoLefrench replied:
i believe i have it intergrated, will test before requesting further pr review

vanyamil commented on Assets/Scripts/UI/HUDManager.cs:
```diff
@@ -0,0 +1,68 @@
+    void Start()
+    {
+        // Assuming PlayerInventory is attached to the player
+        playerInventory = FindObjectOfType<PlayerInventory>();
```
Incorrect comment - you are searching across all game objects for it, attached to player or not doesn't matter here. Consider passing the player or the PlayerInventory in editor, rather than searching across all objects for it.

NickoLefrench replied:
will look into that, that does sound bettter to get it directly

NickoLefrench replied:
Player inventory is now serializable field, added player gameobject to it in inspector

vanyamil commented on Assets/Scripts/UI/HUDManager.cs:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    // Update is called once per frame
+    void Update()
```
Can remove if it is empty

NickoLefrench replied:
will do

NickoLefrench replied:
removed

vanyamil commented on Assets/Scripts/UI/HUDManager.cs:
```diff
@@ -0,0 +1,68 @@
+        }
+
+        // Initialize HUD
+        UpdateDrinkHUD();
```
Put into the null check - then you don't have to check null in the UpdateDrinkHUD function itself.

NickoLefrench replied:
adjusted so that Instead of passing null explicitly, checks if the player has a drink and pass the appropriate Item or null during initialization

vanyamil commented on Assets/Scripts/UI/HUDManager.cs:
```diff
@@ -0,0 +1,68 @@
+            // Get the first drink in the inventory
+            Item drink = playerInventory.itemsList.Find(item => item.IsDrink);
+            
+            //update  drink image
```
You don't need a comment line for each code line. The code lines are pretty self-sufficient/self-descriptive as it is.

NickoLefrench replied:
true that i overcommented, is it a big problem for now?

vanyamil replied:
Better to get good habits early, right

vanyamil commented on Assets/Scripts/Interactables/CocktailMiniGame.cs:
```diff
@@ -259,6 +276,11 @@ private void GiveCorrespondingDrink()
                     //for when we will have the specific sprite icon
                     // itemIcon = GetCocktailIcon(cocktailItemType.Value)
 
+                    //OR?
```
Comment?

NickoLefrench replied:
i left legacy code to see which option worked best, will remove uncecessary stuff x)

NickoLefrench replied:
removed

vanyamil commented on Assets/Scripts/Interactables/CocktailMiniGame.cs:
```diff
@@ -259,6 +276,11 @@ private void GiveCorrespondingDrink()
+                    //OR?
+
+                    // Assign the sprite for the cocktail
+                     itemIcon = cocktailSprite, 
```
Should there be a nullable check?

NickoLefrench replied:
the nullable check should happen, ill double check where i put it

NickoLefrench replied:
null setting#checking is on line 213, 216 as

  //to avoid uninitialized errors
  Item.ItemType? cocktailItemType = null;

  // Placeholder for the cocktail's sprite
  Sprite cocktailSprite = null;

NickoLefrench replied:
If something is set to null beforehand at the beginning of the function, we still need to do a null check before assigning it?

vanyamil replied:
I'm saying that if you pass a possibly null sprite to Item, then there could be no sprite generated when processing the item in the sprite renderer. Should there be a verification somewhere, either in Item class, or whatever uses the itemIcon, that the icon isn't null? Or a verification when putting the icon in the item that it isn't null?

NickoLefrench replied:
added nullable check

vanyamil reviewed (commented):
Could you add a screenshot or recording of what it looks like as part of the review? Since these things will stay in the repository history, it would be nice to record these snapshots.

NickoLefrench committed:
cleanup some comments

NickoLefrench committed:
adjusted as per PR review

set on drink changed as an action event
player inventory is attached through inspector
added nulllable check

vanyamil commented on Assets/Scripts/Inventory/PlayerInventory.cs:
```diff
@@ -17,8 +18,8 @@ public class PlayerInventory : MonoBehaviour
         // Event to notify when a drink is added or removed
-        public delegate void DrinkChanged();
-        public event DrinkChanged OnDrinkChanged;
+       // public delegate void DrinkChanged(); do i still need?
```
can delete

vanyamil commented on Assets/Scripts/UI/HUDManager.cs:
```diff
@@ -10,41 +10,33 @@ public class HUDManager : MonoBehaviour
         {
             // Get the first drink in the inventory
-            Item drink = playerInventory.itemsList.Find(item => item.IsDrink);
+            //Item drink = playerInventory.itemsList.Find(item => item.IsDrink);
```
can delete

vanyamil commented on Assets/Scripts/UI/HUDManager.cs:
```diff
@@ -10,41 +10,33 @@ public class HUDManager : MonoBehaviour
-    {
-        
+        // Initialize HUD with the current state
+        UpdateDrinkHUD(playerInventory.HasDrink() ? playerInventory.itemsList.Find(item => item.IsDrink) : null);
```
I guess it is simpler to write it this way, but in general for optimization, might want to not even call the function with null

NickoLefrench commented:
Uploading TapperRedone - 1stGameplaySnapshot.mp4…

vanyamil commented:
https://github.com/user-attachments/assets/6a3daa27-153f-451f-a824-a91ee527ad53
